cpp/arcticdb/stream/stream_utils.hpp

```diff
@@ -202,56 +202,62 @@ class KeyRangeIterator : public IndexRangeFilter {
 
 inline auto generate_segments_from_keys(
     arcticdb::stream::StreamSource &read_store,
-    std::size_t prefetch_window,
+    ARCTICDB_UNUSED std::size_t prefetch_window,
     const storage::ReadKeyOpts& opts) {
-    using namespace folly::gen;
-    return
-        map([&read_store](auto &&key) {
+
+    return [&read_store, opts](auto&& keys) {
```

Comment:
Returning lambda capturing a ref makes me feel uneasy. I know it was like this before but can you check that we're not trying to use this returned lambda after that reference becomes invalid.